client/src/components/Query/Operator.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Tag, Menu, Dropdown, Icon,
+} from 'antd';
+
+
+export const OPERATOR_TYPE_AND = 'and';
+export const OPERATOR_TYPE_OR = 'or';
+
+
+class Operator extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: null,
+      options: {
+        editable: null,
+      },
+      visible: null,
+    };
+    this.isEditable = this.isEditable.bind(this);
+    this.isVisible = this.isVisible.bind(this);
+    this.serialize = this.serialize.bind(this);
+    this.createMenuComponent = this.createMenuComponent.bind(this);
+    this.handleApply = this.handleApply.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  componentWillMount() {
```

Comment:
`componentWillMount` is deprecated. You probably should do that in the constructor instead.

https://reactjs.org/docs/react-component.html#mounting